openmetadata-service/src/main/java/org/openmetadata/service/resources/EntityResource.java

```diff
@@ -307,6 +306,9 @@ public Response createOrUpdate(UriInfo uriInfo, SecurityContext securityContext,
       entity = addHref(uriInfo, repository.create(uriInfo, entity));
       return new PutResponse<>(Response.Status.CREATED, entity, ENTITY_CREATED).toResponse();
     }
+    resourceContext =
+        getResourceContextByName(
+            entity.getFullyQualifiedName(), ResourceContextInterface.Operation.PUT);
```

Comment:
Should that resource context be set on ln 298 as the entire method will handle `PUT` operations??

Reply:
the method is handling create as well as update. Hence the separation to check CREATE vs EDIT ops. I moved into if else to make it clear